src/openzaak/utils/tasks.py

```diff
@@ -0,0 +1,791 @@
+from openzaak.utils.models import Import, ImportRowResultChoices, ImportStatusChoices
+
+logger = logging.getLogger(__name__)
+
+
+def _get_csv_generator(filename: str) -> Generator[tuple[int, list], None, None]:
+    with open(filename, "r") as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=",", quotechar='"')
+
+        index = 1
+
+        for row in csv_reader:
+            yield index, row
+
+            index += 1
+
+
+def _get_total_count(filename: str, include_header: bool = False) -> int:
+    with open(filename, "r") as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=",", quotechar='"')
+        total = sum(1 for _ in csv_reader)
+
+        if include_header:
+            return total
+
+        return total - 1 if total > 0 else 0
+
+
+@dataclass
+class DocumentRow:
+    _uuid: str
+    _identificatie: str
+    _bronorganisatie: str
+    _creatiedatum: str
+    _titel: str
+    _vertrouwelijkheidaanduiding: str
+    _auteur: str
+    _status: str
+    _formaat: str
+    _taal: str
+
+    _bestandsnaam: str
+    _bestandsomvang: str
+    _bestandspad: str
+
+    _link: str
+    _beschrijving: str
+    _indicatie_gebruiksrecht: str
+    _verschijningsvorm: str
+
+    _ondertekening_soort: str
+    _ondertekening_datum: str
+
+    _integriteit_algoritme: str
+    _integriteit_waarde: str
+    _integriteit_datum: str
+
+    _informatieobjecttype: str
+    _zaak_id: str
+    _trefwoorden: str
+
+    row_index: int
+
+    comment: Optional[str] = None
+    instance: Optional[EnkelvoudigInformatieObject] = None
+
+    _processed: bool = False
+    _succeeded: bool = False
+
+    @classproperty
+    def import_headers(cls) -> list[str]:
+        return [
+            "uuid",
+            "identificatie",
+            "bronorganisatie",
+            "creatiedatum",
+            "titel",
+            "vertrouwelijkheidaanduiding",
+            "auteur",
+            "status",
+            "formaat",
+            "taal",
+            "bestandsnaam",
+            "bestandsomvang",
+            "bestandspad",
+            "link",
+            "beschrijving",
+            "indicatieGebruiksrecht",
+            "verschijningsvorm",
+            "ondertekening.soort",
+            "ondertekening.datum",
+            "integriteit.algoritme",
+            "integriteit.waarde",
+            "integriteit.datum",
+            "informatieobjecttype",
+            "zaakId",
+            "trefwoorden",
+        ]
+
+    @classproperty
+    def export_headers(cls) -> list[str]:
+        return [*cls.import_headers, "opmerking", "resultaat"]
+
+    @property
+    def uuid(self) -> str:
+        if self.instance and self.instance.uuid:
+            return str(self.instance.uuid)
+        return self._uuid
+
+    @property
+    def bronorganisatie(self) -> str:
+        return self._bronorganisatie
+
+    @property
+    def creatiedatum(self) -> Optional[str]:
+        return self._creatiedatum or None
```

Comment:
In this case the `EnkelvoudigInformatieObjectSerializer` requires `creatiedatum` to have an appropriate value (or `None`). The properties are basically added to perform mutations on the provided data without modifying them (so that they can be used later on). I also try not to make fields optional as it might lead to required fields be silently empty without noticing it.